presto-common/src/main/java/com/facebook/presto/common/PrestoVersion.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.common;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static java.util.Objects.requireNonNull;
+
+public class PrestoVersion
+        implements Comparable<PrestoVersion>
+{
+    private static final Pattern VERSION_REGEX = Pattern.compile("^(0\\.([1-9][0-9]*))(\\.([1-9][0-9]*))?(\\.*)?(-([2][0-9]{7}\\.[0-2][0-9]{5}-[0-9]{1,}|SNAPSHOT).*)?$");
```

Comment:
Today our version starts with '0' (i.e. 0.272), doesn't mean it will always stays 0.
The version is split into this section: Major.Minor.Patch where Patch is optional. 
Where,
Major = [0-9]+
Minor = [0-9]+
Patch = [0-9]+
Also we should not check for snapshot version so '-' can be removed.

Reply:
Also do we care if the version has snapshot or not? The version check should be only based on Major and Minor version primarily. wdyt?

Reply:
Yeah, I guess it makes sense since we won't be using the snapshot versions for anything except while testing. 

The only reason I can think of is that "<presto_version-SNAPSHOT" is less that "<presto_version>", so we need to make sure that it is captured.